src/MongoFramework/Attributes/IndexAttribute.cs

```diff
@@ -25,6 +25,10 @@ public class IndexAttribute : Attribute
 		/// The type of index to be applied.
 		/// </summary>
 		public IndexType IndexType { get; }
+		/// <summary>
+		/// Whether the index should add a tenant key.
+		/// </summary>
+		public bool IndexTenant { get; set; }
```

Comment:
Not sure if I like the name - perhaps `IsTenantExclusive`? Thoughts?

Reply:
Went through a couple of different names myself; will change